app/streams/publishing_api_consumer.rb

```diff
@@ -11,24 +11,56 @@ def process(message)
 private
 
   def process_message(message)
-    content_id = message.payload['content_id']
-    base_path = message.payload['base_path']
+    content_id = message.payload.fetch('content_id')
+    base_path = message.payload.fetch('base_path')
+    payload_version = message.payload.fetch('payload_version')
     locale = message.payload['locale'] || 'en'
 
     item = Dimensions::Item.by_natural_key(content_id: content_id, locale: locale)
+
     if item
-      handle_existing(item, base_path, message.delivery_info.routing_key)
+      existing_payload_version = item.publishing_api_payload_version || 0
+
+      if payload_version > existing_payload_version
+        handle_existing(
+          item: item,
+          base_path: base_path,
```

Comment:
Would it make sense to wrap the payload with an object? Something like:

```ruby
class NewVersionEvent
end
```

This would simplify the signature of our methods (there is no need to pass the `content_id`, `base_path`, `locale`...) so many times, and would allow being more explicit in our codebase.

Reply:
I'll think about this. Maybe a new object would help but `NewVersionEvent` is not intuitive to me.

Reply:
I've extracted a new class `PublishingApiMessageProcessor` which the consumer now delegates to.